docs/DeveloperGuide.md

```diff
@@ -7,32 +8,127 @@
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+### Performance Requirements
+- Should work on any mainstream OS as long as it has Java 11 or above installed.
+
+### Usability
+- A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+- Prompts and instructions must be in proper and concise sentences for ease of understanding.
 
 ## Glossary
 
 * *glossary item* - Definition
+* Packages - Travel Package within the agency's database
+* Reservations - Reservation of travel package made by one customer through the app 
 
 ## Instructions for manual testing
 
 {Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+
+### Summary of User Commands
+
+| Command | Format Examples |
+| ------- | --------------- |
+|packages|packages <br/> (prints details of all packages)|
+|info|info {num} (num < number of available packages) <br/> e.g. info 2 |
+|add|add {package_name},{ID},{startDate},{endDate},{hotel},{price},{country},{vacancies} <br/>  e.g. add Skiing Trip,1,23/2/2022,24/2/2022,hotelName,90.99,Singapore,20 <br/> adds a TravelPackage|
+|delete|delete {num} (num < number of available packages) <br/> e.g. delete 2 <br/> delete a TravelPackage|
+|reserve|reserve {package_id},{contact_name},{contact_number},{number_pax} <br/> e.g reserve 3,John,91234567,3|
+|remove|remove {package_id},{contact_number} <br/> e.g remove 1,8888888 <br/> remove an existing reservation|
+|reservations|reservations {package_number} <br/> eg. reservations 2 <br/> print all reservations for a given travelPackageID|
+
+### Test Case 1
+
+New employees can enter `packages` to view all available packages by the agency.
+
+Additionally, they can enter `info {package number}` to view more in-depth information about the package.
+
+![image](https://user-images.githubusercontent.com/64303732/161110520-8ed18ddc-b356-43d5-95ed-7df5ca2de91d.png)
+
+Employees can also input new reservations using `reserve {package_id},{contact_name},{contact_number},{number_pax}`
+
+They can then view and verify reservations for that package using `reservation {package_number}`
+
+![image](https://user-images.githubusercontent.com/64303732/161111482-a2f9600d-d1f0-45b1-9a51-ef28848e4b57.png)
+
+![image](https://user-images.githubusercontent.com/64303732/161111551-c1acfd1e-90f3-426e-b8e3-175f050dba42.png)
+
+
+## Feature - Help Command 
+Aim: Displays a list of all available commands that the user can refer to as a guide
+
+![image](https://user-images.githubusercontent.com/64303732/161110894-f9226a6c-e9f2-4cf3-9158-72cc8df19604.png)
+
+
+
+## Feature - Storage
+#### Initialisation (Loading Data)
+The sequence diagram that shows how `Storage` is created and the data is loaded from the saved files when the program is initialised is shown below: </br>
+![](LoadFileSeqDiag.png)
```

Comment:
1. Missing colon in front of the classes. Refer to the handout for more information on the syntax. 
2. Missing activation bats for calling getPackage(), saveTravelPackage(), instantiating FileWriter and write (text)*.
4. Consider rephrasing the condition for the loop (until all elements are iterated)
5. Consider adding cross to indicate EoL, if any. 

* If you would like to omit the information please ensure that there would not be a loss of data (e.g. can others tell that FileWriter was instantiated in Storage or before?)